GoogleCloudPlatform/microservices-demo
# Move istio-manifests to kustomize component

### Background 
The current [service mesh instructions](https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/docs/service-mesh.md) only include the OSS Istio installation instructions and does not mention the preferred approach for GKE, which is ASM via Fleet API. The instruction also are not using Kustomize components, which is now the preferred approach for demo variants.

### Fixes 
Fixes #1203 

### Change Summary
1. Move istio-manifests to kustomize/components/service-mesh-istio
2. Create test for Istio kustomize component
3. Update hack/make-release-artifacts.sh and kustomize/kustomization.yaml
4. Move instructions into the README on kustomize component

### Additional Notes

Remaining Tasks for this PR:
- [ ] update kustomize/components/service-mesh-istio/README.md
- [ ] update microservices-demo README.md and /docs/cloudshell-tutorial.md
- [ ] incorporate any PR feedback

### Testing Procedure
Install [Anthos Service Mesh](https://cloud.google.com/service-mesh/docs/managed/provision-managed-anthos-service-mesh) or [Istio OSS](https://istio.io/latest/docs/setup/getting-started/), setup an [Istio Ingress Gateway](https://cloud.google.com/service-mesh/docs/gateways), and make sure sidecar injection is enabled in target namespace.

Then from the `kustomize/` folder at the root level of this repository:
```bash
kustomize edit add component components/service-accounts
kustomize edit add component components/service-mesh-istio
kubectl apply -k .
```
